opensoundscape/torch/models/utils.py

```diff
@@ -240,13 +241,22 @@ def tensor_binary_predictions(scores, mode, threshold=None):
         if threshold is None:
             raise ValueError(f"threshold must be specified for multi_target prediction")
         # predict 0 or 1 based on a fixed threshold
-        preds = torch.sigmoid(scores) >= threshold
+        try:
+            if len(threshold) == 1 or len(threshold) == len(scores[0]):
+                preds = torch.sigmoid(scores) >= torch.tensor(threshold)
+            else:
+                raise ValueError(
+                    f"threshold must have the same number of values as there are classes"
+                )
+        except TypeError:  # happens if threshold passed is a float
```

Comment:
handling errors is a logically valid way to handle different input types, but I think its generally better to explicitly handle different cases - the reason is that a type error could potentially occur for a different reason than you expect. So I'd prefer:
```
if type(threshold)==float: 
 ...
 elif len(threshold)==len(classes):
 ....
 else (raise Exception, or maybe you've handled this already)
 ```
 When handling different input possibilities, it can be hard to decipher the code - I will usually add an inline comment at the end of each if: elif: statement eg `#single threshold for all classes...` 
 or `#list of one threshold per class`

Reply:
I was trying to do make the code neater so threshold could be a float, torch.tensor, torch.Tensor, np.array, list or tuple without explicitly type checking for them. - but I think you're right about the risk of catching an unforeseen TypeError.

Reply:
makes sense. I guess the goal is that, if an exception is raised, the user will know what they have to change (without unnecessarily constraining the allowed values). I think the approach I wrote above would handle all the input types you listed, since any non-float type would have to satisfy len()==len(classes)... I guess this isn't quite true, since the type of single value could be np.float32 or something other than float, so the first condition could be more generic than ==float.

Reply:
I've pushed this change, and tried to incorporate all the types it could reasonably be. Though I think it's ugly!